ql/src/java/org/apache/hadoop/hive/ql/ddl/table/execute/AlterTableExecuteAnalyzer.java

```diff
@@ -124,6 +126,11 @@ protected void analyzeCommand(TableName tableName, Map<String, String> partition
       AlterTableExecuteSpec spec =
           new AlterTableExecuteSpec(FAST_FORWARD, new FastForwardSpec(branchName, targetBranchName));
       desc = new AlterTableExecuteDesc(tableName, partitionSpec, spec);
+    } else if (HiveParser.KW_CHERRY_PICK == executeCommandType.getType()) {
+      ASTNode child = (ASTNode) command.getChild(1);
+      long snapshotId = Long.parseLong(child.getText());
+      AlterTableExecuteSpec spec = new AlterTableExecuteSpec(CHERRY_PICK, new CherryPickSpec(snapshotId));
+      desc = new AlterTableExecuteDesc(tableName, partitionSpec, spec);
```

Comment:
I see each `else if` clause has partial same code snippet. e.g.
`ASTNode child = (ASTNode) command.getChild(1);`
and
` desc = new AlterTableExecuteDesc(tableName, partitionSpec, spec);`

Could we extract these common logic to make code more cleaner?

Reply:
I have refactored the code, give a check if it looks better. happy to modify even further if you have suggestions :-)

Reply:
IMO it's good enough. It's more clearly than before. Thx.